### ISSUE TITLE
Lua Coroutine Threads not represented in C#

### ISSUE THREAD

**Issue by AlexanderDahmen** [NLua/NLua#390]
Hello,  
when I use `coroutine.create` in Lua to create a thread, the representation in C# is `object null`.
To get around this, I can use `coroutine.wrap`, which returns a function represented by the `LuaFunction` type in C#.
However, because this is a function and not a thread, I'm not able to call `coroutine.status` on the function wrapping the underlying thread.

Example code:
```c#
using System;
using NLua;

public class Program {
    public static void Main(string[] args) {
        using Lua lua = new Lua();
        object[] result = lua.DoString(@"
            function r(x)
                print('Before yield')
                coroutine.yield()
                print('After yield')
                print('Result: ' .. x)
            end
            return coroutine.create(function() r('Success') end)
        ");

        object thread = result[0]; 
        // Thread is always object null
    }
}
```

I think NLua is missing some type like `LuaThread` to represent coroutine threads properly.
The problem I am facing is the missing ability to check wrapped routines for their status.

Thank you and have a nice day,
Alexander Dahmen

**Comment by viniciusjarina** [comment]
KeraLua uses the Lua object to wrap the Thread state.
Try `lua.State.ToThread(-1)`

**Comment by AlexanderDahmen** [comment]
@viniciusjarina Sorry, what Lua object exactly?
The only object returned is always `null` - how exactly can I access any particular coroutine thread then, if multiple coroutines are active?

**Comment by viniciusjarina** [comment]
I agree this might be a bug the way NLua converts the object from the Lua stack back to the C#
An workaround is to use the KeraLua API (which is a low-lever wrapper on top of Lua C api) to get the Lua thread object. (KeraLua.Lua) using the `ToThread` method (which will call `lua_tothread` )

**Comment by AlexanderDahmen** [comment]
But how should I obtain the ID of the thread exactly, if I'm not able to return the thread from the NLua API?

As for workarounds, I resort to wrapping the LuaFunction generated by `coroutine.wrap` into a C# `struct` containing the LuaFunction, an `alive` boolean and a `Resume` function.
As soon as the LuaFunctions returns a zero-length array on call, the `alive` boolean is set to false.
This is *kind of* like keeping tabs on the routine's status.
